lib/types.ts

```diff
@@ -125,14 +125,57 @@ export class TArray extends TType {
   }
 }
 
+/**
+ * Defines a rest type, e.g. tuple('string', rest(array('number'))).
+ */
+export function rest(typeSpec: TypeSpec): RestType {
+  return new RestType(typeSpec);
```

Comment:
I think it's better to do it like other classes: use `new RestType(parseSpec(typeSpec))` here, and use `TType` as the constructor argument of `RestType`. You won't need the check-for-string in `getChecker` below then.

Reply:
But then the check-for-string just becomes a check for a TName. Ultimately I need to get to a TArray.